internal/service/artifacts/serve.go

```diff
@@ -0,0 +1,119 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/openshift-kni/oran-o2ims/internal/service/artifacts/api"
+	"github.com/openshift-kni/oran-o2ims/internal/service/artifacts/api/generated"
+	common "github.com/openshift-kni/oran-o2ims/internal/service/common/api"
+	"github.com/openshift-kni/oran-o2ims/internal/service/common/clients/k8s"
+)
+
+// Resource server config values
+const (
+	host         = "127.0.0.1"
+	port         = "8000"
+	readTimeout  = 5 * time.Second
+	writeTimeout = 10 * time.Second
+	idleTimeout  = 120 * time.Second
+)
+
+// Serve start artifacts server
+func Serve() error {
+	slog.Info("Starting artifacts server")
+	// Channel for shutdown signals
+	shutdown := make(chan os.Signal, 1)
+	signal.Notify(shutdown, os.Interrupt, syscall.SIGTERM)
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	go func() {
+		sig := <-shutdown
+		slog.Info("Shutdown signal received", "signal", sig)
+		cancel()
+	}()
+
+	// Get client for hub
+	hubClient, err := k8s.NewClientForHub()
+	if err != nil {
+		return fmt.Errorf("error creating client for hub: %w", err)
+	}
+
+	// Init server
+	// Create the handler
+	server := api.ArtifactsServer{
+		HubClient: hubClient,
+	}
+
+	serverStrictHandler := generated.NewStrictHandlerWithOptions(&server, nil,
+		generated.StrictHTTPServerOptions{
+			RequestErrorHandlerFunc:  common.GetOranReqErrFunc(),
+			ResponseErrorHandlerFunc: common.GetOranRespErrFunc(),
+		},
+	)
+
+	router := http.NewServeMux()
+
```

Comment:
one thing that you might need is to add a default route - something like this 
https://github.com/openshift-kni/oran-o2ims/blob/main/internal/service/alarms/serve.go#L123-L125

Reply:
Thank you. Added.